impl/src/main/java/org/jboss/weld/logging/BeanLogger.java

```diff
@@ -538,4 +538,8 @@
 
     @Message(id = 1575, value = "WeldInstance.select(Type subtype, Annotation... qualifiers) can be invoked only on an instance of WeldInstance<Object>.", format = Format.MESSAGE_FORMAT)
     IllegalStateException selectByTypeOnlyWorksOnObject();
+
+    @LogMessage(level = Level.DEBUG)
+    @Message(id = 1576, value = "Using {1} for a shared proxy class {0}; the current impl [{2}] does not match the instantiator the proxy was created with", format = Format.MESSAGE_FORMAT)
```

Comment:
Maybe a little twist?
`Using {1} to instantiate a shared proxy class {0}; former impl [{2}] does not...`

I was also thinking whether we couldn't be more precise and suggest that there are multiple deployments, some of which use relaxed construction while other don't. Or do you think there are other cases where this can happen?
While the log msg tells us something, I guess Average Joe won't have a clue what's happening there.

Reply:
`former` is imho misleading in this use case, but +1 for `to instantiate a shared...`. I don't think we need to be more concrete here. First relaxed construction is an advanced feature. Also it's DEBUG level so I don't expect Average Joe to consume this message.

Reply:
ad `former` - how about `previous`? I just don't like the word `current` as it is current no longer if you see what I mean.

ad. Average Joe - I think even Advanced Joe would have hard times :)
But I am fine leaving it as it is.

Reply:
Hm, `current` means the "configured/default" `ProxyInstantiator` of the current deployment. `previous` is imho also misleading. We don't replace the original instantiator, just obtain a new one to service this particular proxy class.

Reply:
I understand we don't replace it.
Then maybe `this deployment's impl...` or just `default`, `selected`?